### DISCUSSION TITLE
shared x/y axes for multiple views

### DISCUSSION THREAD

**Discussion by manzt** [gosling-lang/gosling.js#630] [Ideas]
Alignment of multiple stacked views is import for accurate comparison of features across tracks. Currently "alignment" of 1D tracks with 2D matrices is accomplished by specifying `xOffset` or `yOffset`. Example:

- `xOffset == 0`
![image](https://user-images.githubusercontent.com/24403730/151445634-81d8ff6d-6626-4009-979e-00eca6aa2690.png)

- `xOffset == 30`
![image](https://user-images.githubusercontent.com/24403730/151445549-28883141-7052-4076-b665-e175800523b9.png)

I think this most often occurs when an axis is specified for a view that is not for other views. Especially since the width and height of a track do not include the space taken by the axes. While I don't think it is necessary, it would be convenient to add this "axis padding" to tracks without requiring users to specify a pixel offset.

**Comment by sehilyi** [comment]
I agree with this "smart default" idea on `xOffset`, and we can think about some potential cases that we want to handle. I think this issue happens mostly with the combination of 1D and 2D views, so we can focus on such cases.

**A. Horizontal View + Matrix View** (Screenshot above)

```js
{ 
  arrangement: "vertical",
  views: [
    { /* View 1: Horizontal View */ },
    { /* View 2: 2D View */ }
  ]
}
```

If both **View 1** and **View 2** have the same `linkingId` (i.e., zooming is linked) and **View 2** shows the y-axis on the left (like the screenshot above), then we can set a certain value to `xOffset` of **View 1** so that both views are left-aligned. Probably, we can additionally check if both **View 1** and **View 2** have the same `width`. If `width` is not the same, then we may not align them.

**B. Vertical View + Horizontal View + Matrix View**

```js
{ 
  arrangement: "horizontal",
  views: [
    { /* View 1: Vertical View */ },
    { 
      arrangement: "vertical",
      views: [
        { /* View 2: Horizontal View */ },
        { /* View 3: 2D View */ }
      ]  
    }
  ]
}
// or
{ 
  arrangement: "vertical",
  views: [
    { /* View 2: Horizontal View */ },
    { 
      arrangement: "horizontal",
      views: [
        { /* View 1: Vertical View */ },
        { /* View 3: 2D View */ }
      ]  
    }
  ]
}
```
<img width="152" alt="Screen Shot 2022-02-01 at 11 47 52" src="https://user-images.githubusercontent.com/9922882/152012424-e6b4b247-4ca5-4ca1-8620-777859c4e53c.png">

If both **View 1**, **View 2**, and **View 3** have the same `linkingId`, then we can left-align the **View 2** and **View 3** and top-align the **View 1** and **View 3**.
